test/testexcludedfiles.cpp

```diff
@@ -42,26 +42,25 @@ class TestExcludedFiles: public QObject
 {
     Q_OBJECT
 
-static auto check_file_full(const char *path)
-{
-    return excludedFiles->fullPatternMatch(path, ItemTypeFile);
-}
-
-static auto check_dir_full(const char *path)
-{
-    return excludedFiles->fullPatternMatch(path, ItemTypeDirectory);
-}
+    static auto check_file_full(const QString &path)
+    {
+        return excludedFiles->fullPatternMatch({ &path }, ItemTypeFile);
+    }
 
-static auto check_file_traversal(const char *path)
-{
-    return excludedFiles->traversalPatternMatch(path, ItemTypeFile);
-}
+    static auto check_dir_full(const QString &path)
+    {
+        return excludedFiles->fullPatternMatch({ &path }, ItemTypeDirectory);
+    }
 
-static auto check_dir_traversal(const char *path)
-{
-    return excludedFiles->traversalPatternMatch(path, ItemTypeDirectory);
-}
+    static auto check_file_traversal(const QString &path)
+    {
+        return excludedFiles->traversalPatternMatch({ &path }, ItemTypeFile);
```

Comment:
I'm not sure you need these braces here. There's no conflicting constructor overloads. It also builds fine on my machine without those.